server/api/api/endpoints/artifacts.py

```diff
@@ -146,15 +146,28 @@ async def get_artifact(
     auth_info: mlrun.common.schemas.AuthInfo = Depends(deps.authenticate_request),
     db_session: Session = Depends(deps.get_db_session),
 ):
-    data = await run_in_threadpool(
-        server.api.crud.Artifacts().get_artifact,
-        db_session,
-        key,
-        tag,
-        iter,
-        project,
-        format_,
-    )
+    try:
+        data = await run_in_threadpool(
+            server.api.crud.Artifacts().get_artifact,
+            db_session,
+            key,
+            tag=tag,
+            iter=iter,
+            project=project,
+            format_=format_,
+        )
+    except mlrun.errors.MLRunNotFoundError:
+        # in earlier versions, producer_id and tag got confused with each other,
```

Comment:
i'm not sure we want to chain the exceptions. We might want to see how the response will look like in case you get 404 for both tries.
I think we want to do something like (see `get_or_create_project` for reasoning):
```suggestion
    not_found = False
    try:
        data = await run_in_threadpool(
            server.api.crud.Artifacts().get_artifact,
            db_session,
            key,
            tag=tag,
            iter=iter,
            project=project,
            format_=format_,
        )
    except mlrun.errors.MLRunNotFoundError:
        not_found = True
     if not_found:
         try:
             ...
          except mlrun.errors.MLRunNotFoundError:
              pass
      if not data:
          raise mlrun.errors.MLRunNotFoundError(some error message)
```

Reply:
The response will be the same as a single try, e.g:
```
<Response [404 Not Found]>
{'detail': "MLRunNotFoundError('Artifact prj/some-key#0:latest not found')"}
```

I can add a log line that we fallback to `tag=tree`, but I don't see why this will happen:
```
 or else the exceptions raised below will be logged along with the "not found" message
```